src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -298,15 +298,25 @@ public static String getIntegerTypeBody() {
    * @return returns the JSON payload to construct a new Type message.
    */
   public static String getBooleanTypeBody() {
-    return getGenericTypeBody("boolean");
+    return getGenericTypeBody("boolean", "");
   }
 
   /**
    * Setup the format for how generic data from this device will be stored with omf
    *
+   * @param type specifies what OMF type the data messages are stored as
+   * @param format specifies what format the type uses on the AF server for bit count and signed or
+   *     unsigned
    * @return returns the JSON payload to construct a new Type message.
    */
-  private static String getGenericTypeBody(String type) {
+  private static String getGenericTypeBody(String type, String format) {
+    String formatString = "";
+
+    // include if not booleans or strings
+    if (type.compareToIgnoreCase("string") != 0 || type.compareToIgnoreCase("boolean") != 0) {
```

Comment:
Couldn't this just be
```suggestion
    if (type.compareToIgnoreCase("string") || type.compareToIgnoreCase("boolean")) {
```